src/components/side-panel/layer-panel/layer-configurator.tsx

```diff
@@ -59,27 +134,27 @@ const StyledLayerVisualConfigurator = styled.div.attrs({
   margin-top: 12px;
 `;
 
-export const getLayerFields = (datasets, layer) =>
-  layer.config && datasets[layer.config.dataId] ? datasets[layer.config.dataId].fields : [];
+export const getLayerFields = (datasets: Datasets, layer: Layer) =>
+  layer.config?.dataId && datasets[layer.config.dataId] ? datasets[layer.config.dataId].fields : [];
 
-export const getLayerDataset = (datasets, layer) =>
-  layer.config && datasets[layer.config.dataId] ? datasets[layer.config.dataId] : null;
+export const getLayerDataset = (datasets: Datasets, layer: Layer) =>
+  layer.config?.dataId && datasets[layer.config.dataId] ? datasets[layer.config.dataId] : null;
 
-export const getLayerConfiguratorProps = props => ({
+export const getLayerConfiguratorProps = (props: LayerConfiguratorProps) => ({
   layer: props.layer,
   fields: getLayerFields(props.datasets, props.layer),
   onChange: props.updateLayerConfig,
   setColorUI: props.updateLayerColorUI
 });
 
-export const getVisConfiguratorProps = props => ({
+export const getVisConfiguratorProps = (props: LayerConfiguratorProps) => ({
   layer: props.layer,
   fields: getLayerFields(props.datasets, props.layer),
   onChange: props.updateLayerVisConfig,
   setColorUI: props.updateLayerColorUI
 });
 
-export const getLayerChannelConfigProps = props => ({
+export const getLayerChannelConfigProps = (props: LayerConfiguratorProps) => ({
```

Comment:
use React.FC

Reply:
This function returns an object with props.